### PR TITLE
acknowledgements

### PR DESCRIPTION
**Pull request by dsavchenko**

### CODE REVIEW

File: odabot/cli.py [hunk 1]
```diff
@@ -504,6 +505,13 @@ def update_workflows(obj, dry_run, force, loop, pattern):
                                                     "running",
                                                     description="Generating frontend tab")
                                     try:
+                                        repo_ls = requests.get(f'{renkuapi}projects/{project["id"]}/repository/tree').json()
+                                        if 'acknowledgements.md' in [x['name'] for x in repo_ls]:
+                                            citation = requests.get(f'{renkuapi}projects/{project["id"]}/repository/files/acknowledgements.md/raw?ref=master').text
+                                            citation = markdown.markdown(citation)
+                                        else:
+                                            citation = f'Service generated from <a href="{project["http_url_to_repo"]}" target="_blank">the repository</a>'
```
[the remark this answers is not in the document]
Comment (dsavchenko) [reply odabot/cli.py:513]: Here is the (rich) text that will to be shown in the frontend panel for the product. 
We may think of adding authors either in the same file (e.g. in the front matter) 

Alternatively, all this info may be specified in the mmoda.yaml. But I went with markdown as it's easier to user to format and preview links etc.

[the remark this answers is not in the document]
Comment (volodymyrss) [reply odabot/cli.py:513]: > Here is the (rich) text that will to be shown in the frontend panel for the product. We may think of adding authors in the same file (e.g. in the front matter)
> 
> Alternatively, all this info may be specified in the mmoda.yaml. But I went with markdown as it's easier for user to format and preview links etc.

Ok sure, this is easier. 
But good to also add list of people from commits.

**Comment by dsavchenko** [comment]
I will try to summarize my understanding

1. Acknowledgements text will be defined in _acknowledgements.md_ as proposed in this PR
2. Maintainer name and email will be defined in _mmoda.yaml_. The logic of this spread between two files is such: _*.md_ is about representation, _mmoda.yaml_ is used for internal configs
4. The list of authors from commits: I don't think we should show it in the acknowledgements field in frontend (or in oda_api response in the future), neither in addition to the citation defined in _acknowledgements.md_ nor in the default placeholder. This list could be quite long. Probably, only maintainer may be shown in the default placeholder.
5. Instead, we use this full list to send email notifications. But we should probably think of the way to unsubscribe.

**Comment by dsavchenko** [comment]
Keeping in mind nb2galaxy conversion, we will need to have citation info in a structured format (only doi or bibtex supported there). Therefore, we need to rethink once again, which file provides which info, to not split related things (this would be confusing for users)  

Still, I prefer to provide an option for the free-text acknowledgements, and we can render bibtex à la galaxy in the future as default

**Comment by volodymyrss** [comment]
> Keeping in mind nb2galaxy conversion, we will need to have citation info in a structured format (only doi or bibtex supported there). Therefore, we need to rethink once again, which file provides which info, to not split related things (this would be confusing for users)
> 
> Still, I prefer to provide an option for the free-text acknowledgements, and we can render bibtex à la galaxy in the future as default

Of course, acknowledgement is not the same thing as citation. The field in the frontend should combine both. Galaxy does not have acknowledgement I guess.

**Comment by dsavchenko** [comment]
As agreed, we will use CITATION.cff

**Comment by dsavchenko** [comment]
> As agreed, we will use CITATION.cff

At least we can try to. But there are restrictions. First, there is no direct support for this format in gitlab (in contrast with github) (but at least it's an issue https://gitlab.com/gitlab-org/gitlab/-/issues/337368), so the created files won't be validated and visualized for the users in renkulab

There is a [cli program](https://github.com/citation-file-format/cffconvert) to convert this format made in python, but it currently lacks support for `preferred-citation` and `references` -- features that are definitely important for scientific workflows. So we will either need to invent the wheel or contribute to that repo. Both are feasible, but may take time, and now we are looking for quick-and-dirty but still functional enough solution. 

There is also a ruby library used by github, but I don't think we want to use ruby

**Comment by volodymyrss** [comment]
Let's just keep acknowledgement text it will be useful anyway.

cff we add later.

**Comment by dsavchenko** [comment]
So let's merge this one. I created an issue to not forget about citations etc.

**Review by volodymyrss — APPROVED** [review]

**Pull request merged**